main.py

```diff
@@ -755,7 +755,13 @@ def handle_link_down(self, event):
                 ):
                     evcs_normal.append(evc)
                     continue
-                for dpid, flows in evc.get_failover_flows().items():
+                try:
+                    dpid_flows = evc.get_failover_flows()
+                except Exception as err:
```

Comment:
Preferred to leave a generic exception handler to catch unexpected errors and avoid the handle link down stops in the middle of its execution.